events/importer/lippupiste.py

```diff
@@ -0,0 +1,371 @@
+        event_time = datetime.strptime(source_event['EventTime'], '%H:%M').time()
+        event_datetime = LOCAL_TZ.localize(datetime.combine(event_date, event_time))
+        event_datetime = event_datetime.astimezone(pytz.utc)
+        event['start_time'] = event_datetime
+        event['provider']['fi'] = source_event['EventPromoterName']
+        event['name']['fi'] = source_event['EventName']
+        event['description']['fi'] = clean_description(source_event['EventSerieText'])
+        event['short_description']['fi'] = clean_short_description(source_event['EventSerieText'])
+        event['info_url']['fi'] = source_event['EventLink']
+        event['image'] = source_event['EventSeriePictureBig_222x222']
+        event['image_license'] = 'event_only'
+
+        existing_keywords = event.get('keywords', set())
+        keywords_from_source = self._get_keywords_from_source_categories(source_event['EventSerieCategories'])
+        event['keywords'] = existing_keywords.union(keywords_from_source)
+
+        place_id = self._get_place_id_from_source_event(source_event)
+        if place_id:
+            event['location']['id'] = place_id
+        else:
+            print("No match found for place '%s' (event %s)" % (source_event['EventVenue'], event['name']['fi']))
+        return event
+
+    def _import_event(self, source_event, events):
+        event_source_id = source_event['EventId']
+        event = events[event_source_id]
+        self._update_event_data(event, source_event)
+
+        # Event and serie IDs separated with namespace, since they may overlap
+        superevent_source_id = get_namespaced_event_serie_id(source_event['EventSerieId'])
+        superevent = events[superevent_source_id]
+        self._update_event_data(superevent, source_event)
+        superevent['origin_id'] = superevent_source_id
+        superevent['super_event_type'] = Event.SuperEventType.RECURRING
+        superevent['info_url']['fi'] = source_event['EventSerieLink']
+
+    def _link_event_to_superevent(self, source_event, events):
+        superevent_source_id = get_namespaced_event_serie_id(source_event['EventSerieId'])
+        try:
+            superevent_id = self.super_event_ids_by_origin_id.get(superevent_source_id)
+        except KeyError:
+            return
+        event_source_id = source_event['EventId']
+        event = events[event_source_id]
+        event['super_event_id'] = superevent_id
+
+    def _update_superevent_details(self, super_event):
+        events = super_event.get_children()
+        if not events.exists():
+            return
+        first_event = events.order_by('start_time').first()
+        super_event.start_time = first_event.start_time
+        super_event.has_start_time = first_event.has_start_time
+        last_event = events.order_by('-end_time').first()
+        super_event.end_time = last_event.end_time
+        super_event.has_end_time = last_event.has_end_time
```

Comment:
Here, it looks like the changes to the start and end time fields are not saved to the superevent, meaning each superevent will have its `start_time` by the *last* subevent, not the first.